src/main/java/competition/subsystems/pose/PoseSubsystem.java

```diff
@@ -31,4 +46,19 @@ protected double getRightDriveDistance() {
         return drive.getRightTotalDistance();
     }
 
+    public Rotation2d getRotation2DGyroAngle() {
+        return Rotation2d.fromDegrees(getCurrentHeading().getValue());
+    }
+
+    @Override
+    public void periodic() {
+        super.periodic();
+        odometry.update(getRotation2DGyroAngle(), Units.inchesToMeters(getLeftDriveDistance()), Units.inchesToMeters(getRightDriveDistance()));
+
+        // updates double property w/ odometry values
+        robotX.set(Units.metersToInches(odometry.getPoseMeters().getTranslation().getX()));
+        robotY.set(Units.metersToInches(odometry.getPoseMeters().getTranslation().getY()));
+        robotHeading.set(odometry.getPoseMeters().getRotation().getDegrees());   
+    }
+
```

Comment:
The `BasePoseSubsystem` has a method called `setCurrentPosition()` that is used to force the robot to a new position on the field. This is typically done to move the robot to a correct starting position for autonomous.

You'll need to override that method, call it's `super` method, and then also do something similar for this new odometry.